wtxmgr/unconfirmed.go

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2013-2015 Conformal Systems LLC <info@conformal.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+package wtxmgr
+
+import (
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/walletdb"
+)
+
+// insertMemPoolTx inserts the unmined transaction record.  It also marks
+// previous outputs referenced by the inputs as spent.
+func (s *Store) insertMemPoolTx(ns walletdb.Bucket, rec *TxRecord) error {
+	v := existsRawUnmined(ns, rec.Hash[:])
+	if v != nil {
+		// TODO: compare serialized txs to ensure this isn't a hash collision?
+		return nil
+	}
+
+	log.Infof("Inserting unconfirmed transaction %v", rec.Hash)
+	v, err := valueTxRecord(rec)
+	if err != nil {
+		return err
+	}
+	err = putRawUnmined(ns, rec.Hash[:], v)
+	if err != nil {
+		return err
+	}
+
+	for _, input := range rec.MsgTx.TxIn {
+		prevOut := &input.PreviousOutPoint
+		k := canonicalOutPoint(&prevOut.Hash, prevOut.Index)
+		err = putRawUnminedInput(ns, k, rec.Hash[:])
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO: increment credit amount for each credit (but those are unknown
```

Comment:
Reminder.  There are several other TODOs, but they're only optimizations.  This is logic related.